### PR TITLE
Additional modifications to the `memory_quota` renaming

### PR DESCRIPTION
**Pull request by adam8157**
Please check the separated commits for details.

### CODE REVIEW

File: src/include/utils/sync_guc_name.h [hunk 1]
```diff
@@ -86,7 +85,6 @@
 		"gp_motion_slice_noop",
 		"gp_quicklz_fallback",
 		"gp_resgroup_debug_wait_queue",
-		"gp_resgroup_memory_policy_auto_fixed_mem",
```
Comment (kainwen) [inline src/include/utils/sync_guc_name.h:89]: `gp_resgroup_memory_policy_auto_fixed_mem` and `gp_resgroup_memory_policy` is also removed???? I think they are still needed.

Comment (SmartKeyerror) [reply src/include/utils/sync_guc_name.h:89]: +1, they're still in using.

File: gpcontrib/gp_toolkit/gp_toolkit--1.3.sql [hunk 2]
```diff
@@ -1735,7 +1735,7 @@ CREATE VIEW gp_toolkit.gp_resgroup_config AS
          , T2.value    AS cpu_max_percent
          , T3.value    AS cpu_weight
          , T4.value    AS cpuset
-         , T5.value    AS memory_quota
+         , T5.value    AS memory_limit
```
Comment (ashwinstar) [inline gpcontrib/gp_toolkit/gp_toolkit--1.3.sql:1738]: This change is just good to have and cosmetic, right? Nothing breaks if the customers don't upgrade the extension and use the new memory_quota for creating the RG grammar.

Comment (adam8157) [reply gpcontrib/gp_toolkit/gp_toolkit--1.3.sql:1738]: If the customers don't upgrade the extension, they need to use the new `memory_quota` but this toolkit function `gp_resgroup_config` shows the column as `memory_limit`.

File: src/include/utils/sync_guc_name.h (hunk 1 above)
Comment (adam8157) [reply src/include/utils/sync_guc_name.h:89]: Forgot @yanwr1 took them back
```
commit c25db0c1e5a897465224efe479d0241a454b673a
Author: wenru yan <ywenru@vmware.com>
Date:   Tue Feb 7 01:39:41 2023 +0000

    implement resource group memory limit control

    1.support create/alter resource group with memory_limit, add the
    removed gucs(which are used to do memory limit) back;
...
```

**Comment by adam8157** [comment]
https://dev.ci.gpdb.pivotal.io/teams/main/pipelines/gpdb-dev-memory_quota-rocky8

**Review by SmartKeyerror — APPROVED** [review]

**Comment by kainwen** [comment]
`gp_resgroup_print_operator_memory_limits` this is useful, I don't remember which commit remove it.
We rely this. If it is not there, we need to add it back.

**Comment by adam8157** [comment]
> `gp_resgroup_print_operator_memory_limits` this is useful, I don't remember which commit remove it. We rely this. If it is not there, we need to add it back.

Removed by:

	commit 483adea86b50c1759460a6265b3d8e3f4198d92e
	Author: Zhenglong Li <lzhenglong@vmware.com>
	Date:   Fri Dec 23 15:36:32 2022 +0800

	    add new semantics on resource group and removing memory model.

Then c25db0c1e5a897465224efe479d0241a454b673a only takes the symbol back, without the GUC codes.

**Pull request merged**